.travis.yml

```diff
@@ -22,6 +22,7 @@ deploy:
     tags: true
     python: 3.6
     branch: master
```

Comment:
is this a contradiction? is says `branch: master` and right below it says `    condition: $TRAVIS_BRANCH =~ ^master$`, which I guess mean `NOT master`

Reply:
I know it is confusing. I had a tag published from PR branch and after reading travis docs about `on` I realized with `tags` entry travis ignores branch name. Apparently they consider tags and branch names mutually exclusive. So until I can update my understanding about the best release approach I decided to add the above condition to make sure nothing will be released from a pr branch.

Reply:
I know it is confusing. I had a tag published from PR branch and after reading travis docs about `on` I realized with `tags` entry travis ignores branch name. Apparently they consider tags and branch names mutually exclusive. So until I can update my understanding about the best release approach I decided to add the above condition to make sure nothing will be released from a pr branch.

Reply:
I know it is confusing. I had a tag published from PR branch and after reading travis docs about `on` I realized with `tags` entry travis ignores branch name. Apparently they consider tags and branch names mutually exclusive. So until I can update my understanding about the best release approach I decided to add the above condition to make sure nothing will be released from a pr branch.

Reply:
Please see: https://docs.travis-ci.com/user/deployment#conditional-releases-with-on